### PR TITLE
[KubeRay][Doc] KubeRay doc migration (3/n)

### PR DESCRIPTION
**Pull request by kevin85421**
## Why are these changes needed?

https://github.com/ray-project/ray/pull/39345

## Related issue number

https://github.com/ray-project/ray/pull/39345

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by architkulkarni — APPROVED** [review]

**Comment by kevin85421** [comment]
cc @angelinalg would you mind reviewing or approving this PR so that we can merge it before the cherry-pick deadline? Thanks!

File: doc/source/cluster/kubernetes/user-guides/k8s-autoscaler.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+### Fine-grained control of scale-down
+To accommodate the statefulness of Ray applications, the Ray autoscaler has more
+fine-grained control over scale-down than the Horizontal Pod Autoscaler. In addition to
+determining desired scale, the Ray Autoscaler is able to select precisely which Pods
+to scale down. The KubeRay operator then deletes that Pod.
+By contrast, the Horizontal Pod Autoscaler can only decrease a replica count, without much
+control over which Pods are deleted. For a Ray application, downscaling a random
+Pod could be dangerous.
+
+### Architecture: One Ray Autoscaler per Ray Cluster.
```
Comment (angelinalg) [inline doc/source/cluster/kubernetes/user-guides/k8s-autoscaler.md:34]:
```suggestion
### Architecture: One Ray Autoscaler per Ray Cluster
```

File: doc/source/cluster/kubernetes/user-guides/k8s-autoscaler.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+## Ray Autoscaler with Kubernetes Cluster Autoscaler
+The Ray Autoscaler and the
+[Kubernetes Cluster Autoscaler](https://github.com/kubernetes/autoscaler/tree/master/cluster-autoscaler)
+complement each other.
+After the Ray autoscaler decides to create a Ray Pod, the Kubernetes Cluster Autoscaler
+can provision a Kubernetes node so that the Pod can be placed.
+Similarly, after the Ray autoscaler decides to delete an idle Pod, the Kubernetes
+Cluster Autoscaler can clean up the idle Kubernetes node that remains.
+It is recommended to configure your RayCluster so that only one Ray Pod fits per Kubernetes node.
+If you follow this pattern, Ray Autoscaler Pod scaling events will correspond roughly one-to-one with cluster autoscaler
```
Comment (angelinalg) [inline doc/source/cluster/kubernetes/user-guides/k8s-autoscaler.md:59]:
```suggestion
If you follow this pattern, Ray Autoscaler Pod scaling events correspond roughly one-to-one with cluster autoscaler
```

**Review by angelinalg — APPROVED** [review]
I just did a very, very quick pass.

File: doc/source/cluster/kubernetes/user-guides/configuring-autoscaling.md [hunk 3]
```diff
@@ -27,200 +28,291 @@ with the KubeRay operator.
...
-2. Workload resource requirements are aggregated by the Ray head container
-   and communicated to the Ray autoscaler sidecar.
-3. The autoscaler determines that a Ray worker pod must be added to satisfy the workload's resource requirement.
-4. The autoscaler requests an additional worker pod by incrementing the RayCluster CR's `replicas` field.
-5. The KubeRay operator creates a Ray worker pod to match the new `replicas` specification.
-6. The Ray scheduler places the user's workload on the new worker pod.
+```{admonition} 3 levels of autoscaling in KubeRay
+  * **Ray actor/task**: Some Ray libraries, like Ray Serve, can automatically adjust the number of Serve replicas (i.e., Ray actors) based on the incoming request volume.
+  * **Ray node**: Ray Autoscaler automatically adjusts the number of Ray nodes (i.e., Ray Pods) based on the resource demand of Ray actors/tasks.
+  * **Kubernetes node**: If the Kubernetes cluster lacks sufficient resources for the new Ray Pods that the Ray Autoscaler creates, the Kubernetes Autoscaler can provision a new Kubernetes node. ***Users must configure this themselves.***
```
[the remark this answers is not in the document]
Comment (kevin85421) [reply doc/source/cluster/kubernetes/user-guides/configuring-autoscaling.md:34]: "this" means the Kubernetes Autoscaler.

[the remark this answers is not in the document]
Comment (kevin85421) [reply doc/source/cluster/kubernetes/user-guides/configuring-autoscaling.md:34]: I will edit your suggestion from "You must configure this new Kubernetes node yourself." to "You must configure the Kubernetes Autoscaler yourself."

**Comment by architkulkarni** [comment]
Windows test_actor_bounded_threads unrelated

**Pull request merged**